CHANGELOG.md

```diff
@@ -41,17 +41,27 @@ Jump in to this release, with the new demos and examples:
 - `UnsupervisedSampler` supports a `walker` parameter to use other random walking algorithms such as `BiasedRandomWalk`, in addition to the default `UniformRandomWalk`. [\#1187](https://github.com/stellargraph/stellargraph/pull/1187)
 - The `StellarGraph` class is now smaller, faster and easier to construct and use:
   - The `StellarGraph(..., edge_type_column=...)` parameter can be used to construct a heterogeneous graph from a single flat `DataFrame`, containing a column of the edge types [\#1284](https://github.com/stellargraph/stellargraph/pull/1284). This avoids the need to build separate `DataFrame`s for each type, and is significantly faster when there are many types. Using `edge_type_column` gives a 2.6× speedup for loading the `stellargraph.datasets.FB15k` dataset (with almost 600 thousand edges across 1345 types).
-  - `StellarGraph`'s internal cache of node adjacencies is now computed lazily [\#1291](https://github.com/stellargraph/stellargraph/pull/1291) and takes into account whether the graph is directed or not [\#1463](https://github.com/stellargraph/stellargraph/pull/1463) (up to 63% smaller, 30× speedup on a reddit dataset with 11.6 million edges), and they now uses the smallest integer type it can [\#1289](https://github.com/stellargraph/stellargraph/pull/1289) (36% smaller on the reddit dataset)
+  - `StellarGraph`'s internal cache of node adjacencies is now computed lazily [\#1291](https://github.com/stellargraph/stellargraph/pull/1291) and takes into account whether the graph is directed or not [\#1463](https://github.com/stellargraph/stellargraph/pull/1463) (up to 63% smaller, 30× speedup on a reddit dataset with 11.6 million edges), and they now use the smallest integer type they can [\#1289](https://github.com/stellargraph/stellargraph/pull/1289) (36% smaller on the reddit dataset)
   - `StellarGraph`'s internal list of source and target nodes are now stored using integer "ilocs" [\#1267](https://github.com/stellargraph/stellargraph/pull/1267), reducing memory use and making some functionality significantly faster (selection of speedups: `StellarGraph.to_adjacency_matrix` 1.6×, GraphSAGE sampling 1.4×, weighted `BiasedRandomWalk` 2.4× [\#1444](https://github.com/stellargraph/stellargraph/pull/1444), `UniformRandomMetaPathWalk` 6× [\#1446](https://github.com/stellargraph/stellargraph/pull/1446))
   - Functions like `graph.node_features()` no longer needs `node_type` specified if `graph` has only one node type (this includes classes like `HinSAGENodeGenerator`, which no longer needs `head_node_type` if there is only one node type) [\#1375](https://github.com/stellargraph/stellargraph/pull/1375)
+- Performance and memory usage improvements in numbers:
```

Comment:
Maybe all of the performance numbers here could be collated slightly, e.g. remove the two reddit dataset and "(selection of speedups: ...)" paratheticals above and put all of the performance numbers here. A possibility might be:

```markdown
- Performance and memory usage improvements in numbers: (or the other title)
  - The FB15k graph has 15 thousand nodes and 483 thousand edges: it is now 7× faster and 4× smaller to construct (without adjacency lists). It is still about 2× smaller when directed or undirected adjacency lists are computed.
  - Directed adjacency matrix construction is 1.6× faster
  - Various samplers and random walkers are faster: `HinSAGENodeGenerator` is 3× faster (on `MovieLens`), `Attri2VecNodeGenerator` is 4× faster (on `CiteSeer`), weighted `BiasedRandomWalk` is up to 3× faster, `UniformRandomMetapathWalk` is up to 7× faster
```

Rendered view:

> - Performance and memory usage improvements in numbers: (or the other title)
>   - The FB15k graph has 15k nodes and 483k edges: it is now 7× faster and 4× smaller to construct (without adjacency lists). It is also about 2× smaller after directed or undirected adjacency lists are computed.
>   - Directed adjacency matrix construction is 1.6× faster
>   - Various samplers and random walkers are faster: `HinSAGENodeGenerator` is 3× faster (on `MovieLens`), `Attri2VecNodeGenerator` is 4× faster (on `CiteSeer`), weighted `BiasedRandomWalk` is up to 3× faster, `UniformRandomMetapathWalk` is up to 7× faster